cmd/monaco/cmdutils/cmdutils.go

```diff
@@ -34,30 +36,56 @@ func SilenceUsageCommand() func(cmd *cobra.Command, args []string) {
 
 var ssoTokenURL = "https://sso.dynatrace.com/sso/oauth2/token" //nolint:gosec
 
-// VerifyClusterGen takes a manifestEnvironments map and tries to call the version endpoint of each environment
-// in order to verify that the user has configured the environments correctly.
-// Depending on the configured environment "type" the function tries to call the version endpoint of either
-// classic gen or platform gen. The function will return an error as soon as
-// it receives an error from calling the version endpoint of an environment
-func VerifyClusterGen(envs manifest.Environments) error {
+// VerifyEnvironmentGeneration takes a manifestEnvironments map and tries to verify that each environment can be reached
+// using the configured credentials
+func VerifyEnvironmentGeneration(envs manifest.Environments) bool {
 	for _, env := range envs {
 		switch env.Type {
 		case manifest.Classic:
-			if _, err := client.GetDynatraceVersion(client.NewTokenAuthClient(env.Auth.Token.Value), client.Environment{URL: env.Url.Value, Type: client.Classic}); err != nil {
-				return fmt.Errorf("could not verify Dynatrace cluster generation of environment %q (%q). Please check the configured Auth credentials in the manifest", env.Name, env.Url)
-			}
+			return isClassicEnvironment(env)
 		case manifest.Platform:
-			oauthCredentials := client.OauthCredentials{
-				ClientID:     env.Auth.OAuth.ClientId.Value,
-				ClientSecret: env.Auth.OAuth.ClientSecret.Value,
-				TokenURL:     ssoTokenURL,
+			return isPlatformEnvironment(env)
+		default:
+			log.Error("Could not authorize against the environment with name %q (%s). Unknown environment type.")
+			return false
+		}
+	}
+	return true
+}
+
+func isClassicEnvironment(env manifest.EnvironmentDefinition) bool {
+	if _, err := client.GetDynatraceVersion(client.NewTokenAuthClient(env.Auth.Token.Value), env.Url.Value); err != nil {
+		var respErr client.RespError
+		if errors.As(err, &respErr) {
+			log.Error("Could not authorize against the environment with name %q (%s) using token authorization.", env.Name, env.Url.Value)
+			if respErr.StatusCode != http.StatusForbidden {
+				log.Error("Please verify that this environment is a Dynatrace Classic environment.")
+			} else {
+				log.Error(err.Error())
 			}
-			if _, err := client.GetDynatraceVersion(client.NewOAuthClient(oauthCredentials), client.Environment{URL: env.Url.Value, Type: client.Platform}); err != nil {
-				return fmt.Errorf("could not verify Dynatrace cluster generation of environment %q (%q). Please check the configured Auth credentials in the manifest", env.Name, env.Url)
+		}
```

Comment:
We should probably also log something if the error is not a Response Error? 
E.g. I think that could happen when the URL is just completely wrong/there's no server replying?

Maybe a more general "Could not connect to environment" and printing the error.
Currently we swallow the actual error if it's not a RespErr

Reply:
an error in this case is already ultimately printed to the user ... looks smth like that:
```
2023/03/17 08:53:02 ERROR HTTP Request failed with Error: Get "https://WRONG.dev.apps.dydnatracelabs.com/api/v1/config/clusterversion": dial tcp: lookup WRONG.dev.apps.dydnatracelabs.com on 1.2.3.4 no such host
Error: unable to verify Dynatrace environment generation
```
I'll try to enhance it a bit